rust/ip-packet/src/ipv4_header_slice_mut.rs

```diff
@@ -7,15 +7,12 @@ pub struct Ipv4HeaderSliceMut<'a> {
 
 impl<'a> Ipv4HeaderSliceMut<'a> {
     /// Creates a new [`Ipv4HeaderSliceMut`].
-    ///
-    /// # Safety
-    ///
-    /// - The byte array must be at least of length 20.
-    /// - The IP version must be 4.
-    pub unsafe fn from_slice_unchecked(slice: &'a mut [u8]) -> Self {
```

Comment:
I decided to remove these unsafe functions because I don't think it adds anything. It is just a couple if `if`s!

Reply:
agreed with this, we can readd them if the benchmarks tell us we're spending a lot of time with these operations.

Reply:
Please also audit the other unsafe functions. Those are much more local so I think it should be fine :)